x-pack/plugin/esql/qa/server/src/main/java/org/elasticsearch/xpack/esql/qa/rest/RestEsqlTestCase.java

```diff
@@ -1007,34 +1020,77 @@ public static Map<String, Object> runEsqlAsync(RequestObjectBuilder requestObjec
     }
 
     static Map<String, Object> runEsql(RequestObjectBuilder requestObject, AssertWarnings assertWarnings, Mode mode) throws IOException {
+        return runEsql(requestObject, assertWarnings, mode, null, null);
+    }
+
+    static Map<String, Object> runEsql(
+        RequestObjectBuilder requestObject,
+        AssertWarnings assertWarnings,
+        Mode mode,
+        @Nullable String format,
+        @Nullable Character delimiter
+    ) throws IOException {
         if (mode == ASYNC) {
-            return runEsqlAsync(requestObject, assertWarnings);
+            return runEsqlAsync(requestObject, assertWarnings, format, delimiter);
         } else {
-            return runEsqlSync(requestObject, assertWarnings);
+            return runEsqlSync(requestObject, assertWarnings, format, delimiter);
         }
     }
 
     public static Map<String, Object> runEsqlSync(RequestObjectBuilder requestObject, AssertWarnings assertWarnings) throws IOException {
+        return runEsqlSync(requestObject, assertWarnings, null, null);
+    }
+
+    public static Map<String, Object> runEsqlSync(
+        RequestObjectBuilder requestObject,
+        AssertWarnings assertWarnings,
+        @Nullable String format,
+        @Nullable Character delimiter
+    ) throws IOException {
         requestObject.build();
         Request request = prepareRequest(SYNC);
         String mediaType = attachBody(requestObject, request);
 
         RequestOptions.Builder options = request.getOptions().toBuilder();
         options.setWarningsHandler(WarningsHandler.PERMISSIVE); // We assert the warnings ourselves
         options.addHeader("Content-Type", mediaType);
-
+        boolean isTextFormat = format != null;
         if (randomBoolean()) {
-            options.addHeader("Accept", mediaType);
+            if (isTextFormat) {
+                switch (format) {
+                    case "txt" -> options.addHeader("Accept", "text/plain");
+                    case "csv" -> options.addHeader("Accept", "text/csv");
+                    case "tsv" -> options.addHeader("Accept", "text/tab-separated-values");
+                }
+            } else {
+                options.addHeader("Accept", mediaType);
+            }
```

Comment:
If this needs to be duplicated, build a static map of text formats to Accept values and then you can compact this (and elsewhere) to smth like:
```
options.addHeader("Accept", TEXT_FORMATS.getOrDefault(mediaType, mediaType));
```
And similar in the alternate branch.

However, it might be clearer if you made `runEsqlAsTextWithFormat()` take a format and add implementation for async (potentially still refactoring, to reuse code, where it makes sense), rather than carry on the format&delimiter around and then wrap the textual results into a map just to conform to the return type of the existing non-text methods.